container-core/src/main/java/com/yahoo/container/jdisc/HttpResponse.java

```diff
@@ -129,4 +125,13 @@ public void complete() { }
         return Collections::emptyIterator;
     }
 
+    /** Sets the type classification of this request for metric collection purposes */
+    public void setRequestType(Request.RequestType requestType) { this.requestType = requestType; }
```

Comment:
Why is request type as member of `HttpResponse`? I would assume `HttpRequest` is more suitable as the "type" is a property of the request, not the response. It's also stored in the request type for the jdisc core abstraction layer.

Reply:
That allows the handler/processors/searchers to process the request and decide the right classification based on its content.

Reply:
Jdisc filters cannot perform this classification easily if it's not part of the request type.

Reply:
Well there's lots of things filters cannot perform easily - they are special purpose. Is there a good use case for filters doing this classification?

Reply:
Access control